### DISCUSSION TITLE
Boundary conditions at the interface when a fluid flows vertically over a solid surface

### DISCUSSION THREAD

**Discussion by sychang2021** [idaholab/moose#21466] [Q&A Modules: Navier-Stokes] [answered]
Hello everyone!

I am trying to solve a problem of liquid flowing in a vertical pipe. There is an obstacle in the pipe, and the liquid flows in the area between the pipe and the obstacle. But I can't solve the correct pressure field and velocity field. I think there may be a problem with the velocity boundary conditions on the surface of the liquid and the obstacle. Shouldn't the boundary conditions of the fluid and solid interface be Neumann boundary conditions, but other boundary conditions? Are there any similar cases in MOOSE that I can learn from?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Does the pipe only case work as expected? Are you getting the expected pressure drop then?
What discretization are you using for Navier Stokes for this problem? FE? INS? INSAD?

Guillaume

**Reply by sychang2021** [reply]
INSAD. I have found the problem. I forgot to add no-slip boundary conditions to the walls of solids and fluids. Adding this boundary condition gives reasonable results. thank you for your reply.
